addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -702,11 +704,25 @@ def _do_reverse_moves(self):
         )
         draft_moves.unlink()
 
+    def _compute_default_accounting_date(self, most_recent_expense):
+        if most_recent_expense == fields.Date.context_today(self):
+            return most_recent_expense
+        else:
+            start_month = fields.Date.start_of(fields.Date.context_today(self), "month")
+            end_month = fields.Date.end_of(fields.Date.context_today(self), "month")
+            if start_month <= most_recent_expense <= end_month:
+                return fields.Date.context_today(self)
+            elif most_recent_expense < start_month:
+                lock_date = self.company_id._get_user_fiscal_lock_date()
+                if lock_date == date.min:
+                    return fields.Date.end_of(fields.Date.start_of(fields.Date.context_today(self), "year"), "month")
+                else:
+                    return min(fields.Date.context_today(self), fields.Date.end_of(fields.Date.add(lock_date, months=1), "month"))
```

Comment:
Here is a bunch of tips to make it easier to read:
- we could save `fields.Date.context_today(self)` in a variable `context_today` or `today` to make it thinner and much natural to read
- As the first `if` has a `return`, we can remove the `else` clause and disindent all the rest one level
- We don't really consider the case in which `most_recent_expense > end_month` (because it has no sense that today > today_end_of_month) and thus 
```py
if start_month <= most_recent_expense <= end_month:
``` 
could be:
```py
if start_month <= most_recent_expense:
```
- Also, then we don't need the end_month anymore and we can delete the variable
- after that `elif most_recent_expense < start_month:` is basically just the opposite of the `elif` and thus can be an `else`
- as the if has a return too, the else can be remove and we can disindent the rest
- The last else can also be removed and the last line disindent